project/src/main/kotlin/org/ionproject/core/calendar/CalendarData.kt

```diff
@@ -0,0 +1,286 @@
+            val summary = rs.getSummaries(SUMMARIES_COLUMN)
+            val description = rs.getDescriptions(DESCRIPTIONS_COLUMN)
+            val dtStamp = DateTimeStamp(rs.getDatetime(DTSTAMP_COLUMN))
+            val created = DateTimeCreated(rs.getDatetime(CREATED_COLUMN))
+
+            return when (type) {
+                "E" -> Event(
+                    uid,
+                    summary,
+                    description,
+                    dtStamp,
+                    created,
+                    categories,
+                    DateTimeStart(rs.getDatetime(DTSTART_COLUMN)),
+                    DateTimeEnd(rs.getDatetime(DTEND_COLUMN)),
+                    rs.getRecurrenceRule(BYDAY_COLUMN)
+                )
+                "J" -> Journal(
+                    uid,
+                    summary,
+                    description,
+                    rs.getAttachments(ATTACHMENTS_COLUMN),
+                    dtStamp,
+                    DateTimeStart(rs.getDatetime(DTSTART_COLUMN)),
+                    created,
+                    categories
+                )
+                "T" -> Todo(
+                    uid,
+                    summary,
+                    description,
+                    rs.getAttachments(ATTACHMENTS_COLUMN),
+                    dtStamp,
+                    created,
+                    DateTimeDue(rs.getDatetime(DUE_COLUMN)),
+                    categories
+                )
+                else -> throw UnknownCalendarComponentTypeException("The type represented by $type is not known.")
+            }
+        }
+
+        private fun ResultSet.getSummaries(columnName: String): Array<Summary> {
+            val summaries = getArray(columnName).array as Array<String>
+
+            return summaries.map { summary ->
+                val text = summary.substringAfter(':')
+                val language = summary.substringBefore(':').toInt()
+
+                Summary(
+                    text,
+                    language = languageRepo.byId(language)
+                )
+            }.toTypedArray()
+        }
+
+        private fun ResultSet.getDescriptions(columnName: String): Array<Description> {
+            val descriptions = getArray(columnName).array as Array<String>
+
+            return descriptions.map { description ->
+                val text = description.substringAfter(':')
+                val language = description.substringBefore(':').toInt()
+
+                Description(
+                    text,
+                    language = languageRepo.byId(language)
+                )
+            }.toTypedArray()
+        }
+
+        private fun ResultSet.getCategories(columnName: String): Array<Categories> {
+            val tempCats = getArray(columnName).array as Array<java.lang.Integer>
+
+            val cats = tempCats.map { it.toInt() }
+
+            return cats.map {
+                categoryRepo.byId(it) ?: TODO("FOREIGN KEY EXCEPTION")
+            }.groupBy {
```

Comment:
Imo, filter/query related operations should be handled by the SQL queries alone. SQL is optimized to run such functions in one go.

Reply:
This call is not getting anything from the database.